### DISCUSSION TITLE
VRF Chainlink takes long time for response and fail

### DISCUSSION THREAD

**Discussion by SebastianContrerasR** [Cyfrin/foundry-full-course-cu#1504] [Q&A]
Hello, I've been testing the Raffle.sol Testnet Demo with VRF and Chainlink Automation. It's taking a long time, and I'm not sure why this is happening or if I've done something wrong.

I followed the steps in the video, but the VRF is taking a considerable amount of time. It's been 17 minutes so far, and there's still no response from the VRF. I wanted to know if this is normal or not.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/55514936/f5f74091-4bd9-48f3-9c40-2cccf5b22426)

**Comment by EngrPips** [comment]
Hello @Santana20 , This is not normal as i didn't wait up to two minutes when i did that section of the course to get back a response from the `VRF`, I have heard someone run into this issue when they requested two words from the `VRF` but i am not very sure why is that. How many words did you requested in your request ?

**Reply by SebastianContrerasR** [reply]
I think is 1.
This is the contract. https://sepolia.etherscan.io/address/0x21cc24b9c9401fad17c242b5ab04df2760eacfb3#code

**Comment by SebastianContrerasR** [comment]
i think is vrf chainlink problem, because. I tested with the [foundry-smart-contract-lottery](https://github.com/Cyfrin/foundry-smart-contract-lottery-f23) and i have the same problem.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/55514936/9160a51d-605c-4033-bd3d-1d8a7a1c30fe)

**Reply by SebastianContrerasR** [reply]
This is the VRF: https://vrf.chain.link/sepolia/10395

**Comment by EngrPips** [comment]
Hello @Santana20, So i am not very sure what is wrong and why it is taking so long for you to get back a random number from the `VRF`, I did tested with my already deployed lottery [deployed to sepolia chain] contract and i got back a random number within few seconds not even minutes , below are pictures.

<img width="1437" alt="Screenshot 2024-03-21 at 2 14 23 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/d48f4733-086d-4db0-917f-50ffec8f7ca1">

<img width="1440" alt="Screenshot 2024-03-21 at 2 15 53 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/306961c2-9c24-49aa-9d93-eeeb46df7fdd">

Are you testing on sepolia test net ?

**Reply by SebastianContrerasR** [reply]
Hi, yes, on sepolia test net. This is the [VRF details](https://vrf.chain.link/sepolia/10395)

**Reply by EngrPips** [reply]
Hello @Santana20 , so i clicked on the hash of the transaction so as to see the state of the transaction and it showed that it has been successful picture below 

<img width="1440" alt="Screenshot 2024-03-21 at 7 32 54 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/1f29d28e-d515-4a71-bfe7-865fea8a51f4">

I would advice that you check the state of your contract and confirm that indeed the random number had been received and necessary modification of the contract state had taken place

**Reply by SebastianContrerasR** [reply]
Hi, well I checked if the contract state changed. In the picture below we see the contract has a calculating state and has 1 player yet.
![Screenshot_2024-03-21-09-57-58-590_com.brave.browser.jpg](https://github.com/Cyfrin/foundry-full-course-f23/assets/55514936/98d98764-5f5b-4123-8b64-3b536242bd74)

**Comment by SebastianContrerasR** [comment]
Hi, I checked the [VRF ](https://vrf.chain.link/sepolia/10395)again and it shows me that the call failed. Has something like this happened to anyone?
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/55514936/79d13182-9bc9-4747-bcbf-81881dc80d28)
